lib/conduit_async_ssl.ml

```diff
@@ -45,15 +44,27 @@ let ssl_connect ?version ?name ?ca_file ?ca_path ?session ?verify
   >>= fun conn ->
   verify_connection conn >>= function
   | false ->
+    Ssl.Connection.close conn ;
+    Pipe.close_read app_rd ;
+    Writer.close w >>= fun () ->
```

Comment:
How come you're closing the reader in 1 case and the pipe in the other? Seems a bit asymmetrical.